### PR TITLE
Allow auth to be activated with a robot user admin

### PR DESCRIPTION
**Pull request by msteffen**
This means that it's possible to start a pachyderm cluster and enable auth without any human interaction.  However, when auth is activated, Pachyderm returns an auth token authenticated the caller as the robot admin, and if that token is lost everyone will be locked out of that cluster

### CODE REVIEW

File: src/server/auth/cmds/cmds_test.go [hunk 1]
```diff
@@ -188,9 +191,82 @@ func TestModifyAdminsPropagateError(t *testing.T) {
...
+		tu.Cmd("pachctl", "auth", "modify-admins", "--add=admin").Run())
+}
+
+func TestActivateMismatchedUsernames(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	// We need a custom 'activate' command, so reproduce 'activateAuth' minus the
+	// actual call
+	activateMut.Lock()
```
Comment (sjezewski) [inline src/server/auth/cmds/cmds_test.go]: Seems like this could be made into a helper like `idempotentEnterpriseActivate()`

File: src/server/auth/server/admin_test.go [hunk 2]
```diff
@@ -1257,3 +1257,44 @@ func TestGetAuthTokenErrorNonAdminUser(t *testing.T) {
 	require.YesError(t, err)
 	require.Matches(t, "must be an admin", err.Error())
 }
+
+// TestActivateAsRobotUser tests that Pachyderm can be activated such that the
+// initial admin is a robot user (i.e. without any human intervention
```
Comment (sjezewski) [inline src/server/auth/server/admin_test.go]: Typo ... missing end paren

File: src/server/pfs/server/driver.go [hunk 3]
```diff
@@ -209,6 +209,15 @@ func absent(key string) etcd.Cmp {
...
 func (d *driver) createRepo(ctx context.Context, repo *pfs.Repo, description string, update bool) error {
+	// Check that the user is logged in (user doesn't need any access level to
+	// create a repo, but they must be authenticated if auth is active)
+	whoAmI, err := d.pachClient.AuthAPIClient.WhoAmI(auth.In2Out(ctx),
+		&auth.WhoAmIRequest{})
+	authIsActivated := !auth.IsErrNotActivated(err)
+	if authIsActivated && err != nil {
+		return fmt.Errorf("error authenticating (must log in to create a repo): %v",
+			grpcutil.ScrubGRPC(err))
+	}
```
Comment (sjezewski) [inline src/server/pfs/server/driver.go]: I don't think there's a test case for this behavior included. We should have a test for this change.

File: src/server/pps/server/api_server.go [hunk 4]
```diff
@@ -2662,10 +2662,15 @@ func (a *apiServer) getPachClient() *client.APIClient {
 			panic(fmt.Sprintf("pps failed to initialize pach client: %v", err))
 		}
 		// Initialize spec repo
-		if err := a.pachClient.CreateRepo(ppsconsts.SpecRepo); err != nil {
-			if !isAlreadyExistsErr(err) {
-				panic(fmt.Sprintf("could not create pipeline spec repo: %v", err))
+		if err := a.sudo(a.pachClient, func(superUserClient *client.APIClient) error {
```
Comment (sjezewski) [inline src/server/pps/server/api_server.go]: It seems like we're covering an edge case here, but I'm not sure exactly what. Is it that before the spec repo would've used non su creds to create the internal spec repo? Which could fail if ... ?

Is it worth writing a test case for this?

File: src/server/auth/cmds/cmds_test.go (hunk 1 above)
Comment (msteffen) [reply src/server/auth/cmds/cmds_test.go]: Done

File: src/server/auth/server/admin_test.go (hunk 2 above)
Comment (msteffen) [reply src/server/auth/server/admin_test.go]: Fixed

File: src/server/pfs/server/driver.go (hunk 3 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Done (it's not really intended to be a change in semantics, but I think it basically is; this error used to get returned in the STM below)

File: src/server/pps/server/api_server.go (hunk 4 above)
Comment (msteffen) [reply src/server/pps/server/api_server.go]: The change here is actually sort of minor. Basically, this code tries to create a spec repo, and if it already exists, this code ignores the error and continues.

After this PR, if you try to create a repo without logging in, and the repo also already exists, you get `auth token not found`, which breaks this code even though the repo also already exists.

Wrapping this in sudo guarantees that if pachd starts after auth has been activated (which pretty much means that the spec repo has already been created, though there are no hard checks in the auth server that the spec repo exists) then pachd gets a `__spec__ already exists` error here and happily ignores it, rather than an `auth token not found` error which breaks here.

I think this code is actually tested by virtue of the fact that there are auth tests that start pipelines, and each pipeline starts a sidecar which tries to create the spec repo. If this weren't wrapped in sudo, every sidecar that started after auth was activated would crash.

Comment (msteffen) [reply src/server/pps/server/api_server.go]: Just confirmed, this is covered by existing auth tests (when I revert this delta, they fail)
